src/common.rs

```diff
@@ -99,6 +108,33 @@ pub fn get_restful_parameters(urlstring: &str) -> HashMap<&str, &str> {
     parameters
 }
 
+/*
+ * Input: path directory to be changed owner to root
+ * Return: Result contains execution result
+ *         - directory name for successful execution
+ *         - -1 code for failure execution.
+ *
+ * If privilege requirement is met, change the owner of the path to root
+ * This function is unsafely using libc. Result is returned indicating
+ * execution result.
+ */
+pub fn chownroot(path: String) -> Result<String, i32> {
+    unsafe {
+        if libc::geteuid() != 0 {
+            error!("Privilege level unable to change ownership to root for file: {}", path);
+            return Err(-1);
+        } else {
```

Comment:
Close.  Use more early exit  The flow should be like:

```rust
if getuid() != 0 {
    return;
}
if chown() != 0 {
    return;
}
Ok()
```

This keeps the branching manageable, which is easier to read and reason about.